server/controllers/middlewares/errorHandle/error.js

```diff
@@ -0,0 +1,8 @@
+/* eslint-disable no-unused-vars */
+exports.client = (req, res) => {
```

Comment:
@Mu7ammadAbed I think you need all 4 arguments for express to understand that this is an error handling callback, for example `exports.clientError = (err, req, res, next) => {`, otherwise express will just treat it as a normal middleware.

Actually, I think in your current design, both clientError and serverError will both be called for all error conditions since they're both used as middlewares.  Did you intend for them to be used like middlewares, or explicitly imported and called inside your request handlers?

Reply:
This is a middleware to handle 404 missing page or wrong route and the other middleware is for handling the server errror that got thrown using the next(err) function

Reply:
I get it now! Sorry about that, and thanks for explaining.
Does the 404 handler (clientError) need to be the *last* route in the router file for that to work properly?  If so, it might be a good idea to add a comment near it in `controllers/index.js` saying "do not move this, it needs to be below all of the other routes" to protect against it being moved in the future.
Whenever I have code that only works if it's in a certain place in a file or function, I add a comment so people (or my future self) don't accidentally move it into the wrong place.